rpmio/macro.c

```diff
@@ -1174,6 +1177,89 @@ static void doShescape(MacroBuf mb, rpmMacroEntry me, ARGV_t argv, size_t *parse
     mbAppend(mb, '\'');
 }
 
+static unsigned long getmem_total(void)
+{
+    unsigned long mem = 0;
+    long int pagesize = sysconf(_SC_PAGESIZE);
+    long int pages = sysconf(_SC_PHYS_PAGES);
+
+    if (pagesize < 0)
+	pagesize = 4096;
+    if (pages > 0)
+	mem = pages * pagesize;
+
+    return mem;
+}
+
+static unsigned long getmem_proc(int thread)
+{
+    unsigned long mem = getmem_total();
+    /*
+     * Conservative estimates for thread use on 32bit systems where address
+     * space is an issue: 2GB for bare metal, 3GB for a 32bit process
+     * on a 64bit system.
+     */
+    if (thread) {
+	unsigned long vmem = mem;
+#if __WORDSIZE == 32
+	vmem = UINT32_MAX / 2;
+#else
+#if defined(__linux__)
+	if ((personality(0xffffffff) & PER_MASK) == PER_LINUX32)
+	    vmem = (UINT32_MAX / 4) * 3;
+#endif
+#endif
+	/* Fixup to get nice even numbers */
+	vmem = vmem / (1024*1024) + 1;
```

Comment:
This needs to be outside of the block so that the function also returns the no. of megabytes when called with `thead` 0. Right now, it returns the number of bytes which makes the `%{getncpus:proc}` calculation, erm, a bit too optimistic :smile:

Reply:
While at it, maybe add the `proc` variant to the test case, too?

Reply:
Oops :laughing:

Reply:
Fixed + test added in last push.